src/extension-runners/firefox-android.js

```diff
@@ -384,6 +390,23 @@ export class FirefoxAndroidExtensionRunner {
     await adbUtils.amForceStopAPK(selectedAdbDevice, selectedFirefoxApk);
   }
 
+  async adbArtifactsDir(clear: boolean) {
+    const {
+      adbUtils,
+      selectedAdbDevice,
+    } = this;
+
+    const check = await adbUtils.checkOrCleanArtifacts(selectedAdbDevice,
+                                                       clear);
+    if (check) {
+      log.warn(
+        'Artifacts directory found ' +
+        'run something like web-ext --adb-clean-artifacts' +
+        ' to remove older Artifacts'
+      );
+    }
+  }
```

Comment:
I think that here we should only log the warning if the adbUtils method is only checking if any old artifacts directory is present (and it was not actually removing them in this run).